microsoft/testsuites/osu/osusuite.py

```diff
@@ -0,0 +1,164 @@
+        priority=1,
+    )
+    def run_osu_bench(self, node: Node, log: Logger, log_path: Path) -> None:
+
+        _install_osu_mpi(node)
+        path = _install_osu_bench(node)
+
+        gpu_count = node.capability.gpu_count
+        gpu = node.features[Gpu]
+        expected_count = gpu.get_gpu_count_with_lspci()
+        assert_that(gpu_count).described_as(
+            "Expected device count didn't match actual device count from lspci!"
+        ).is_equal_to(expected_count)
+
+        nproc = node.tools[Lscpu].get_core_count()
+        tests = [
+            "allgather",
+            "allreduce",
+            "alltoall",
+            "bcast",
+            "gather",
+            "reduce_scatter",
+            "reduce",
+            "scatter",
+        ]
+        for test in tests:
+            node.execute(
+                f"mpirun -np {nproc} python3.8 {path}/python/run.py"
+                f" --benchmark {test} --min 128 --max 10000000 >> "
+                f"{path}/osu_bench.log",  # 2>&1
+                expected_exit_code=0,
+                expected_exit_code_failure_message=(
+                    f"failed to run mpirun {test} benchmark"
+                ),
+                timeout=1200,
+                shell=True,
+                sudo=True,
+            )
+        log.info("OSU MPI tests finish!")
+        osu_log_path = str(node.get_pure_path(f"{path}/osu_bench.log"))
+        node.tools[Chmod].chmod(osu_log_path, "a+rwX", sudo=True)
+        node.shell.copy_back(
+            node.get_pure_path(osu_log_path), PurePath(log_path) / "osu_bench.log"
+        )
+
+
+def _install_osu_mpi(node: Node) -> None:
+
+    assert isinstance(node.os, Posix)
+    node.os.install_packages("libibverbs-dev gfortran bison")
+    wget = node.tools[Wget]
+    tar = node.tools[Tar]
+
+    path = node.get_working_path()
+    if node.shell.exists(path / OSU_MPI_FILE_NAME):
+        return
+
+    download_path = wget.get(
+        url=OSU_MPI_LOCATION, filename=str(OSU_MPI_FILE_NAME), file_path=str(path)
+    )
+    tar.extract(download_path, dest_dir=str(path))
+
+    source_code_folder_path = PurePosixPath(f"{path}/mvapich2-2.3.7-1")
+    node.execute(
+        "./configure --prefix=/usr/local/",
+        cwd=source_code_folder_path,
+        shell=True,
+        expected_exit_code=0,
+        expected_exit_code_failure_message="fail to run configure command",
+    )
+    node.tools[Make].make_install(cwd=source_code_folder_path)
+
+
+def _install_osu_bench(node: Node) -> str:
+
+    wget = node.tools[Wget]
+    tar = node.tools[Tar]
+
+    path = node.get_working_path()
+    if node.shell.exists(path / OSU_BENCH_FILE_NAME):
+        return str(path) + "/osu-micro-benchmarks-6.2"
+
+    download_path = wget.get(
+        url=OSU_BENCH_LOCATION, filename=str(OSU_BENCH_FILE_NAME), file_path=str(path)
+    )
+    tar.extract(download_path, dest_dir=str(path))
+    assert isinstance(node.os, Posix)
+    node.os.install_packages("python3.8-dev")
```

Comment:
change it into `node.os.install_packages("python3.8-dev python3.8-distutils")`, others are good!